src/plugin/page-generation/utils.ts

```diff
@@ -0,0 +1,236 @@
+  const padding = options?.padding | 20;
+  const previousElement = options?.previousElement;
+
+  const frame = figma.createFrame();
+  frame.name = name;
+  frame.layoutMode = 'HORIZONTAL';
+  frame.layoutAlign = 'STRETCH';
+  frame.itemSpacing = padding;
+  frame.paddingLeft = padding;
+  frame.paddingRight = padding;
+  frame.paddingTop = padding;
+  frame.paddingBottom = padding;
+
+  if (previousElement) {
+    frame.y = previousElement.y + previousElement.height + 50;
+  }
+
+  return frame;
+};
+
+/**
+ * Assign to each slice the matching
+ * ref ids using setSharedPluginData
+ */
+export const setRefs = ({
+  slices,
+  refIds,
+}: {
+  slices: readonly SceneNode[] | DefaultShapeMixin[];
+  refIds: RefIds[Slices];
+}) => {
+  slices.map((slice) => {
+    if (refIds?.hasOwnProperty(slice.name)) {
+      slice.setSharedPluginData(PLUGIN_DATA_NAMESPACE, slice.id, refIds[slice.name]);
+    }
+  });
+};
+
+/**
+ * Takes the radiiFrame as parameter and sync
+ * matching Box variants with current style settings
+ *
+ * It returns the existing and new variants added
+ */
+export const syncRadiiVariants = (radiiFrame: FrameNode) => {
+  const newRadiusSlices: Record<string, number> = {};
+  const existentRadiusSlices: Record<string, number> = {};
+
+  radiiFrame.children.map((radiiSlice) => {
+    if (radiiSlice.type !== 'RECTANGLE') {
+      return;
+    }
+
+    const refIds = radiiSlice.getSharedPluginData(PLUGIN_DATA_NAMESPACE, radiiSlice.id).split('/#/');
+    refIds.map((refId) => {
+      const boxVariant = figma.getNodeById(refId);
+
+      if (radiiSlice.cornerRadius === figma.mixed) {
+        figma.notify('Mixed border radius is not allowed on Slices', { error: true });
+        return;
+      }
+
+      if (!boxVariant && existentRadiusSlices[radiiSlice.name]) {
+        figma.notify(`The ${radiiSlice.name} variant on Radii already exist, please use a unique name`, {
+          error: true,
+          timeout: 2000,
+        });
+        return;
+      }
+
+      if (!boxVariant) {
+        newRadiusSlices[radiiSlice.name] = radiiSlice.cornerRadius;
+        return;
+      }
+
+      if (boxVariant.type === 'COMPONENT') {
+        existentRadiusSlices[radiiSlice.name] = radiiSlice.cornerRadius;
+        boxVariant.cornerRadius = radiiSlice.cornerRadius;
+        boxVariant.name = boxVariant.name.replace(/Radius=.*?,/, `${Slices.Radius}=${radiiSlice.name},`);
+      }
+    });
+  });
+
+  return { existentRadiusSlices, newRadiusSlices };
+};
+
+/**
+ * Takes the borderWidthsFrame as parameter and sync
+ * matching Box variants with current style settings
+ *
+ * It returns the existing and new variants added
+ */
+export const syncBorderWidthVariants = (borderWidthsFrame: FrameNode) => {
+  const newBorderWidthSlices: Record<string, number> = {};
+  const existentBorderWidthSlices: Record<string, number> = {};
+
+  borderWidthsFrame.children.map((borderWidthSlice) => {
+    if (borderWidthSlice.type !== 'LINE') {
```

Comment:
I don't think this is important now, but we can consider allowing the user to also use the rectangle to set this design token

Reply:
Not sure about that, it'll add some complexity (a few checks to understand what type is the slice and how to behave for each case). But even if from UX perspective could be good (less restrictive rules), IMO for now we should decide just a single way and enforce it.

Looks like this topic should be discussed, could you open a dedicated issue?

Reply:
Sounds good, I'm gonna open the issue to discuss about it